core/common/src/DisplayStyleSettings.ts

```diff
@@ -518,7 +520,10 @@ export class DisplayStyleSettings {
     this._monochromeMode = MonochromeMode.Flat === this._json.monochromeMode ? MonochromeMode.Flat : MonochromeMode.Scaled;
 
     this._backgroundMap = BackgroundMapSettings.fromJSON(this._json.backgroundMap);
-    this._mapImagery = MapImagerySettings.fromJSON(this._json.mapImagery, this._json.backgroundMap);
+
+    // TODO
+    // Should we have a migration path here with the old JSON
```

Comment:
Absolutely. Saved views created before this change using old version of BackgroundMapProps must continue to load and display as they did previously.

Reply:
Sent you an email related to this, I need advice on how this should be correctly done.  One problem is that `this._json`  is typed with the 'new' definition, and previous properties are not accessible.  Is casting to `any` and retrieve legacy settings a good way to solve this?

Reply:
We need to draw a distinction between changing the API and changing the persistence format. Every saved view in existence today stores their background map settings using BackgroundMapProps, so we are stuck supporting that forever - we can only add to the persistence format, not subtract. IMO you should mark BackgroundMapProps.providerName/Data as `@deprecated` (that term refers to an API, not to data) and `@internal` but not remove them.